### PR TITLE
chore: Move peerDependencies to dependencies

### PR DESCRIPTION
**Pull request by benoitdeziel**
## Description

Our previous setup for our config was assuming that the user would install the dependencies by themselves but this could cause some frustration when trying to stay on top of every package updates.

This PR moves all the `peerDependencies` to `dependencies` so that it will be installed with the config package. Only the tools like `eslint`, `prettier` and `stylelint` are kept as `peerDependencies`.

All the dependencies have been updated to their latest version.

**I think that we should bump to `1.0` for all configs with this release.**

### Before

Screen Shot 2022-03-08 at 09 57 27

### After

Screen Shot 2022-03-08 at 09 56 05

## Checklist

- [x] Test any new configuration on a repo using [`npm link`](https://docs.npmjs.com/cli/link)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install eslint-config-landr@0.9.0-canary.93.41dbfe7.0
  npm install prettier-config-landr@0.4.0-canary.93.41dbfe7.0
  npm install stylelint-config-landr@0.4.0-canary.93.41dbfe7.0
  # or 
  yarn add eslint-config-landr@0.9.0-canary.93.41dbfe7.0
  yarn add prettier-config-landr@0.4.0-canary.93.41dbfe7.0
  yarn add stylelint-config-landr@0.4.0-canary.93.41dbfe7.0
  ```
</details>

### CODE REVIEW

**Review by eyeballquip — APPROVED** [review]

**Review by hjack-landr — APPROVED** [review]

**Comment by hjack-landr** [comment]
accidentally approved this from my non-LANDR account 😅

**Comment by benoitdeziel** [comment]
@hjack-landr I was wondering who this was for a sec 😆

File: .eslintrc.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module.exports = {
-    extends: ['landr'],
+    extends: ['./packages/eslint-config-landr'],
```
Comment (davidmartinezmarin) [inline .eslintrc.js:2]: are you telling me that the linting was not working properly on the linting repo ? 🙄

**Review by davidmartinezmarin — APPROVED** [review]
Noice 🤌

**Comment by davidmartinezmarin** [comment]
Haakon are you trying to bypass the 2 reviews approval with your 2 accounts ? 🤣  @hjack-landr @eyeballquip

File: .eslintrc.js (hunk 1 above)
Comment (benoitdeziel) [reply .eslintrc.js:2]: It was working but the thing is that it was taking the one on npm, now it uses directly the one in the repo

File: packages/stylelint-config-landr/package.json [hunk 2]
```diff
@@ -16,14 +16,14 @@
     ],
     "main": "index.js",
```
Comment (ardeois) [inline packages/stylelint-config-landr/package.json:17]: Are we still using stylelint?

**Review by ardeois — APPROVED** [review]
LGTM
I'll test it on a project in local and will let you know

File: packages/stylelint-config-landr/package.json (hunk 2 above)
Comment (benoitdeziel) [reply packages/stylelint-config-landr/package.json:17]: We should for any styled-components

**Comment by benoitdeziel** [comment]
@hjack-landr @davidmartinezmarin @ardeois I saw that in some project we're using an eslint plugin for jest, should we add it to this repo?

**Comment by eyeballquip** [comment]
@benoitdeziel I think it's a good idea - we also use the `testing-library` eslint plugin in Samples

**Comment by benoitdeziel** [comment]
:rocket: PR was released in [`eslint-config-landr@1.0.0`](https://github.com/LandrAudio/linting-and-formatting/releases/tag/eslint-config-landr%401.0.0), [`prettier-config-landr@1.0.0`](https://github.com/LandrAudio/linting-and-formatting/releases/tag/prettier-config-landr%401.0.0), [`stylelint-config-landr@1.0.0`](https://github.com/LandrAudio/linting-and-formatting/releases/tag/stylelint-config-landr%401.0.0) :rocket:

**Pull request merged**